### PR TITLE
Add support for 1.8 stable

### PR DESCRIPTION
**Pull request by misohu**
**Added**:
- support for 1.8 stable 

**Changes**:
- Set juju version based of the chosen Kubeflow version
- Add fix for juju 3.1 setup 
- Change Kubernetes versions for beta nad edge 

**Tested**: 
- tested al lversions on my AWS account

**How to review**:
- You can try to deploy the stack based on the steps in REAMDE.md

### CODE REVIEW

File: cf_kubeflow_single_instance.yaml [hunk 1]
```diff
@@ -124,18 +140,21 @@ Resources:
             apt update
             apt -y upgrade
             for snap in juju-wait charmcraft; do sudo snap install $snap --classic; done
-            sudo snap install juju --classic --channel=2.9/stable
+            sudo snap install juju --classic --channel=${jujuVersion}/stable
```
Comment (kimwnasptd) [inline cf_kubeflow_single_instance.yaml:143]: How is a default value getting assigned here?

Comment (kimwnasptd) [reply cf_kubeflow_single_instance.yaml:143]: Is it via the 
```
jujuVersion: !FindInMap [JujuVersionMap, !Ref KubeflowVersion, jujuVersion]
```
line below?

Comment (misohu) [reply cf_kubeflow_single_instance.yaml:143]: yes i will use the KubeflowVersion's default value to set the juju default value.

**Review by NohaIhab — COMMENTED** [review]
Tested it, lgtm.
I was following the testing instructions in the README.md and they were only missing filling the `KubeflowDashboardPassword` parameter.
You can add it to the readme or rather just link the [tutorial](https://charmed-kubeflow.io/docs/deploy-kubeflow-with-aws-cloud-formation) from our docs, the instructions seem to be a duplicate of those.

File: cf_kubeflow_single_instance.yaml [hunk 2]
```diff
@@ -124,18 +140,21 @@ Resources:
...
             chown -f -R ubuntu /home/ubuntu/.kube
+            chown -f -R ubuntu /home/ubuntu/.local/share
 
             microk8s enable dns storage metallb:"10.64.140.43-10.64.140.49,192.168.0.105-192.168.0.111"
-            sleep 120
+            sleep 60
             microk8s.kubectl wait --for=condition=available -nkube-system deployment/coredns deployment/hostpath-provisioner
             microk8s.kubectl -n kube-system rollout status ds/calico-node
 
+            su ubuntu -c 'microk8s config | juju add-k8s microk8s --client'
```
Comment (kimwnasptd) [inline cf_kubeflow_single_instance.yaml]: Last nit, let's also use a different K8s name here from `microk8s` :)
(and on the command right next)

**Comment by misohu** [comment]
Hi @NohaIhab thanks for review I added the note to README.md ... we need to keep it duplicated as the aws requires git repo with readme when we submit it.

**Review by NohaIhab — APPROVED** [review]
LGTM

**Pull request merged**